### PR TITLE
[JSDK-48] Text editing

### PR DESCRIPTION
**Pull request by KamillaGarifullina**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
-* All in one place: here you can find everything you need to build modern and reliable web application with Uploadcare on board
-* Modern technologies: web components, service workers, wasm, and other cutting edge web standards are ready to show their power
-* Type safety with a JSDoc declarations and TypeScript static analysis
-* Modern ESM-level code sharing
-* Build stage is not necessary for development: use the code directly in any modern browser or node.js
-* Fast and secure
-* Minimum of external dependencies: security audit is not expensive
-* We trying to be close to native web platform as possible and do not to force you to use any specific development environment tools
-* Tested by Uploadcare community
+## 💎 Core concept
```
Comment (rsedykh) [inline README.md:3]: concepts?

File: README.md [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
+* Minimum of external dependencies: security audit is not expensive.
+* We are trying to be close to the native web platform as possible and don't force you to use any specific development environment tools.
+* Tested by Uploadcare community.
 
 ## 📦 What’s in the box?
 
-* [upload-blocks](./upload-blocks/) - build your own uploading flow with the simple HTML and CSS
-* Adaptive image delivery tools: all you need to distribute your web images efficiently
-* File uploading client library: handle files from browser or node
+* [upload-blocks](./upload-blocks/) - build your own uploading flow with the simple HTML and CSS.
```
Comment (rsedykh) [inline README.md:17]: Please, typography. —, not -.

File: README.md [hunk 3]
```diff
@@ -1,39 +1,39 @@
...
-* Analytic tools: get your page score
-* Common purpose components and tools
-* etc...
+* Cloud image editing tools: tune your pictures on-the-fly.
+* Analytic tools: get your page score.
+* Common purpose components and tools.
+* etc.
 
-## ⚙️ Installation and startup:
+## ⚙️ Installation and startup
```
Comment (rsedykh) [inline README.md:26]: startup?

File: upload-blocks/README.md [hunk 4]
```diff
@@ -1,40 +1,42 @@
 # &lt;upload-blocks&gt;
 
-## 🧩 Bulld you own file uploading flow with the set of pre-defined custom elements!
+## 🧩 Build your own file uploading flow with the set of pre-defined custom elements!
```
Comment (rsedykh) [inline upload-blocks/README.md:3]: ! at the end?..

File: upload-blocks/README.md [hunk 5]
```diff
@@ -1,40 +1,42 @@
...
 > Or dive deeper and create your own beautiful blocks!
 
 ## 🍰 Concept description
 
-There are so many use cases and many workflows for file uploading... Is it possible to create uploading solution to fit them all? 
+There are so many use cases and many workflows for file uploading.
+Is it possible to create an uploading solution to fit them all? 
 
-We believe - yes, with a power of [Custom Elements standard](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) and our widget-purpose-specific open source [Symbiote.js](https://github.com/symbiotejs/symbiote.js) library.
+We believe - yes, with a power of [Custom Elements standard](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) and our widget-purpose-specific open-source [Symbiote.js](https://github.com/symbiotejs/symbiote.js) library.
```
Comment (rsedykh) [inline upload-blocks/README.md:12]: open-source? Where have you found it?

File: upload-blocks/README.md [hunk 6]
```diff
@@ -1,40 +1,42 @@
...
+We use [Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) standard to make integration seamless. That allows us to use simple HTML code to define layouts and place our widgets into any other templates or markup. Custom Elements are compatible with most modern technologies and provide a consistent workflow for all of them. Please, check out the list of high-level tests passed with a set of modern frontend frameworks and libraries: [Custom Elements Everywhere](https://custom-elements-everywhere.com/)
 
-Also, we providing the set of [reference integrations](https://github.com/uploadcare/upload-blocks-examples).
+We also provide a set of [reference integrations](https://github.com/uploadcare/upload-blocks-examples).
 
-To integrate Uploadcare widget, first of all, you should connect our script:
+To integrate the Uploadcare widget, first of all, you should connect our script:
```
Comment (rsedykh) [inline upload-blocks/README.md:39]: We should definitely ask Natasha to review such changes.  Or to have another reviewer. I can provide one.

File: upload-blocks/README.md [hunk 7]
```diff
@@ -43,31 +45,34 @@ To integrate Uploadcare widget, first of all, you should connect our script:
...
   style="--pubkey:'demopublickey'; --darkmode: 1"
   css-src="../upload-blocks/themes/uc-basic/index.css">
 </uc-default-widget>
 ```
+
 Some of widget settings in this example are passed via CSS variables. That means you can use styles, CSS-classes and dedicated CSS-files to pass any setting to any block or redefine if using native DOM API or just native HTML syntax. Let's move forvard to the next section, to explain that.
 
+Some of the widget settings in this example are passed via CSS variables. That means you can use styles, CSS classes, and dedicated CSS files to give any setting to any block or redefine if using native DOM API or just native HTML syntax. Let's move forward to the next section to explain that.
+
 ## 🎨 Customize everything!
```
Comment (rsedykh) [inline upload-blocks/README.md:72]: !!!11

File: upload-blocks/README.md [hunk 8]
```diff
@@ -43,31 +45,34 @@ To integrate Uploadcare widget, first of all, you should connect our script:
...
 ```html
 <uc-default-widget
   style="--pubkey:'demopublickey'; --darkmode: 1"
   css-src="../upload-blocks/themes/uc-basic/index.css">
 </uc-default-widget>
 ```
+
 Some of widget settings in this example are passed via CSS variables. That means you can use styles, CSS-classes and dedicated CSS-files to pass any setting to any block or redefine if using native DOM API or just native HTML syntax. Let's move forvard to the next section, to explain that.
 
+Some of the widget settings in this example are passed via CSS variables. That means you can use styles, CSS classes, and dedicated CSS files to give any setting to any block or redefine if using native DOM API or just native HTML syntax. Let's move forward to the next section to explain that.
```
Comment (rsedykh) [inline upload-blocks/README.md:70]: Somebody forget to clean up after.

**Review by rsedykh — CHANGES_REQUESTED** [review]
Didn't have strength to review it properly, but still. (And I'll read it thoroughly later).

File: upload-blocks/README.md (hunk 5 above)
Comment (KamillaGarifullina) [reply upload-blocks/README.md:12]: Here https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/o/open-source

File: README.md (hunk 1 above)
Comment (paninebaluy) [reply README.md:3]: concepts

File: upload-blocks/README.md (hunk 8 above)
Comment (paninebaluy) [reply upload-blocks/README.md:70]: Also, what did you mean by "native HTML syntax?" Is this about using default browser styling?

Comment (paninebaluy) [reply upload-blocks/README.md:70]: "... to give any setting to any block or redefine if using native DOM API or just native HTML syntax".
Blocks usually have CSS properties, not settings. And they may be overridden by [styles with greater specificity](https://www.w3schools.com/css/css3_variables_overriding.asp).
I suggest this rewrite of the second sentence:

That means you can use _(inline -- please clarify)_ styles, CSS classes, and dedicated CSS files to give any property to any block or override styles if your project uses native DOM API _(or just native HTML syntax -- to clarify)_.

File: upload-blocks/README.md [hunk 9]
```diff
@@ -43,31 +45,34 @@ To integrate Uploadcare widget, first of all, you should connect our script:
...
+
 ## 🎨 Customize everything!
 
-First of all, please take a look at [our CSS aproach discussion](https://github.com/uploadcare/jsdk/discussions/18). Don't be shy to participate, your opinion is very important for us.
+First of all, please look at [our CSS approach discussion](https://github.com/uploadcare/jsdk/discussions/18). 
+Please don't be shy to participate; your opinion is very important for us.
```
Comment (paninebaluy) [inline upload-blocks/README.md:75]: Please don't hesitate to take part; your opinion is very important to us.

File: upload-blocks/README.md [hunk 10]
```diff
@@ -91,19 +96,21 @@ All basic configurations for each block could be provided via the set of [CSS-va
...
+
+Any configuration value can be defined and redefined at any level of the DOM tree at any time.
 
 ### Look & feel
 
-Each block has a reference CSS file, placed at the common directory with the JavaScript class defenition module. You can use it as the template to create your onw custom styling (or CSS-animations) for the any library block. All blocks and their children are acting like a regular DOM-elements in that case, you can use native CSS-selectors and the any of the well known styling approaches. All significant component states are provided as the element attribute flags, so you can use the full power of CSS to customize blocks in deep.
+Each block has a reference CSS file, placed at the common directory with the JavaScript class definition module. You can use it as the template to create your own custom styling (or CSS-animations) for any library block. All blocks and their children act like regular DOM elements; in that case, you can use native CSS-selectors and any of the well-known styling approaches. All significant component states are provided as the element attribute flags, so you can use the full power of CSS to customize blocks in deep.
```
Comment (paninebaluy) [inline upload-blocks/README.md:107]: Each block has a reference CSS file that's located in a directory together with a JavaScript class definition module. You can use it as the template to create your own custom styling (or CSS animations) for any library block. All blocks and their children act like regular DOM elements; therefore, you can use native CSS selectors and any existing styling approach. All significant component states are provided as element attribute flags, so you can use the full power of CSS to deeply customize the blocks.

File: upload-blocks/README.md [hunk 11]
```diff
@@ -91,19 +96,21 @@ All basic configurations for each block could be provided via the set of [CSS-va
...
 
 [More about CSS theme creation]()
 
 ### Layout & composition
 
-Block-components can be used separately or in combinations. You can compose them into the common workflow:
+Block components can be used separately or in combinations. You can compose them into the common workflow:
```
Comment (paninebaluy) [inline upload-blocks/README.md:113]: You can combine them to create a common workflow like this one:

File: upload-blocks/README.md (hunk 5 above)
Comment (paninebaluy) [inline upload-blocks/README.md:12]: We believe it is &mdash; with the power of [Custom Elements standard](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) and our widget-purpose-specific open-source [Symbiote.js](https://github.com/symbiotejs/symbiote.js) library.

File: README.md [hunk 12]
```diff
@@ -1,39 +1,39 @@
...
-* Tested by Uploadcare community
+## 💎 Core concept
+
+* All in one place: here you can find everything you need to build a modern and reliable web application with Uploadcare on board.
+* Modern technologies: web components, service workers, wasm, and other cutting edge web standards are ready to show their power.
+* Type safety with a JSDoc declaration and TypeScript static analysis.
+* Modern ESM-level code sharing.
+* Build stage is not necessary for development: use the code directly in any modern browser or Node.js.
+* Fast and secure.
+* Minimum of external dependencies: security audit is not expensive.
```
Comment (paninebaluy) [inline README.md:11]: * Minimum external dependencies, which makes the security audit really inexpensive.

File: README.md [hunk 13]
```diff
@@ -1,39 +1,39 @@
...
-* Build stage is not necessary for development: use the code directly in any modern browser or node.js
-* Fast and secure
-* Minimum of external dependencies: security audit is not expensive
-* We trying to be close to native web platform as possible and do not to force you to use any specific development environment tools
-* Tested by Uploadcare community
+## 💎 Core concept
+
+* All in one place: here you can find everything you need to build a modern and reliable web application with Uploadcare on board.
+* Modern technologies: web components, service workers, wasm, and other cutting edge web standards are ready to show their power.
+* Type safety with a JSDoc declaration and TypeScript static analysis.
```
Comment (paninebaluy) [inline README.md:7]: * Type safety with the JSDoc declaration and the TypeScript static analysis.

File: README.md [hunk 14]
```diff
@@ -1,39 +1,39 @@
...
+* Modern ESM-level code sharing.
+* Build stage is not necessary for development: use the code directly in any modern browser or Node.js.
+* Fast and secure.
+* Minimum of external dependencies: security audit is not expensive.
+* We are trying to be close to the native web platform as possible and don't force you to use any specific development environment tools.
```
Comment (paninebaluy) [inline README.md:12]: * It's as close to your native web platform as possible and doesn't force you to use any environment-specific development tools.

File: README.md [hunk 15]
```diff
@@ -1,39 +1,39 @@
...
+* Tested by Uploadcare community.
```
Comment (paninebaluy) [inline README.md:13]: * Tested by the Uploadcare community.

File: README.md [hunk 16]
```diff
@@ -1,39 +1,39 @@
...
+## ⚙️ Installation and startup
 
 1. Clone repository: `git clone git@github.com:uploadcare/jsdk.git ./`
 2. Run dependencies setup: `npm run setup`
 3. Start local dev-server: `npm run dev` 
 
 **Or you can use any other local server at the project root**
 
-All code in this repository works without any additional build process in any modern browser, so you can just open any HTML file in project structure to make it work during development or for the testing purposes. No source maps or rebuilds are needed.
+All code in this repository works without any other build process in any modern browser. You can open any HTML file in the project structure to make it work during development or testing purposes. No source maps or rebuilds are needed.
```
Comment (paninebaluy) [inline README.md:34]: You can open any HTML file in the project structure to make it work during the development or testing phase.

File: upload-blocks/README.md (hunk 4 above)
Comment (paninebaluy) [reply upload-blocks/README.md:3]: ## 🧩 Use predefined custom elements to build your own file uploading flow

**Review by paninebaluy — CHANGES_REQUESTED** [review]
There are some questions about the content of this readme. See comments.

**Review by foxeyes — APPROVED** [review]

**Pull request merged**